### ISSUE TITLE
Mismatching series as movie

### ISSUE THREAD

**Issue by vankasteelj** [Ivshti/video-name-parser#3]
https://github.com/Ivshti/video-name-parser/blob/master/index.js#L138

For example: `awesome.show.2014.217.hdtv-source[team].mp4`

When "Awesome Show" is the version of 2014, because it existed previously as a movie or a series, eg: in 1978.

It detects:

```js
Object {
  year: 2014
  name: "awesome show"
  type: "movie"
}
```

Instead of:

```js
Object {
  name: "awesome show 2014"
  type: "series"
  episode: [17]
  season: 2
}
```

There must be some kind of way, instead of ignoring whole because a year was found, check for episode, and if `year && (!episode || !season)` then => movie

**Comment by Ivshti** [comment]
Yes, there is:
https://github.com/Ivshti/video-name-parser/blob/master/index.js#L138-L163

But as you see, it's been disabled, because otherwise creates too many false positives in movies. For example, if a movie is named "the awesome movie 440"

Possible solution would be to see if the stamp is before/after the year

**Comment by vankasteelj** [comment]
well currently, `the awesome movie 440.avi` gives episode 40, season 4 of "the awesome movie", with type="series"

x)

**Comment by Ivshti** [comment]
> well currently, the awesome movie 440.avi

By "named" I meant the movie name itself. The file name would be ``the awesome movie 440 (2014)`` or something like that, in which case it would be indexed as a movie.

I'm OK with a PR for some flags that change that behaviour, and I don't have a very strong opinion on what exactly is right here.

This module has went through a lot of evolution, that doesn't mean all the logic is right, but considering what it's doing there's hardly a "right" way to do things.

So I propose making it a flag

**Comment by vankasteelj** [comment]
There sure is no right way: https://github.com/butterproject/butter-desktop/blob/master/src/app/common.js#L250

I'll dig a little

**Comment by vankasteelj** [comment]
There's this:

```js
        /* Since this one is risky, do it only if we haven't matched a year (most likely not a movie) or if year is BEFORE stamp, like: show.2014.801.mkv */
        if (stamp && (!meta.year || (meta.year && (firstNameSplit.indexOf(stamp.toString()) < firstNameSplit.indexOf(meta.year.toString())))))
        {
            meta.episode = [parseInt(stamp.slice(-2), 10)];
            meta.season = parseInt(stamp.slice(0, -2), 10);
        }
```
here: https://github.com/Ivshti/video-name-parser/blob/master/index.js#L158-L162

**Comment by vankasteelj** [comment]
will you publish to npm?

**Comment by Ivshti** [comment]
@vankasteelj published, 1.2.0

**Comment by vankasteelj** [comment]
neato! thx
